### ISSUE TITLE
Segfault in Qt while using SQLite cipher plugin

### ISSUE THREAD

**Issue by bezhin** [devbean/QtCipherSqlitePlugin#5]
Hello.
I was trying to use the plugin on Ubuntu 16.04 x64 (it is a VM in VirtualBox 5.0.26) and I've got some strange behavior. My application crashes with SIGSEGV with a following stack trace:

```
1 ??                                                                      0x7ffff608b3a9 
2 QSqlCachedResultPrivate::cleanup()                                      0x7ffff608c3b2 
3 QSqlQuery::prepare(QString const&)                                      0x7ffff607d4ae 
4 BB::EventCacheComponent::EventCacheComponent eventcachecomponent.cpp 43 0x439a46       
5 BB::Agent::Agent                             agent.cpp               33 0x412416       
6 main                                         main.cpp                54 0x41090a       
```

Namespace "BB::" corresponds to my app. Levels 1-3 of stack trace correspond to the "libQt5Sql.so" module.
I don't think that the problem is with my code: my app works without any issue if I change database driver to the "QSQLITE".
I tried to run test project from your repository with the same result - it crashed. Here is the stack trace:

```
1 ??                                             0x7fcb4fb673a9 
2 QSqlCachedResultPrivate::cleanup()             0x7fcb4fb683b2 
3 QSqlQuery::exec(QString const&)                0x7fcb4fb58d25 
4 main                               main.cpp 21 0x4019be       
```

Building and running the same app on Windows 10 x64 does not trigger this issue (msvc2013 x64 and x86 builds work fine). Also, there was a report about similar issue with my app on Mac OS X some time ago, though I'm not sure it is related. Qt version: 5.7.0 on all platforms. Does anyone know what exactly is the problem and how to fix it?

**Comment by vasvlad** [comment]
I confirmed this bug for QT 5.7. In the version QT 5.6 it works fine.

**Comment by devbean** [comment]
Would you please send me the test code copy via email (devbean#outlook.com)? I have tested using a simple application on Qt 5.7 OpenSUSE gcc 64 and it works fine.

**Comment by vasvlad** [comment]
I've fixed using 5.7 including files. See my patch
https://github.com/vasvlad/QtCipherSqlitePlugin/commit/3b55f41ab58586389b92935e44ee42b6558b96a4

**Comment by devbean** [comment]
@vasvlad Yes, the code from Qt 5.7 has changed a lot. That's why I decided to copy code from Qt private to this plugin -- in order to avoid changing our code a lot when Qt changes. If you copy from Qt directly,  everyone is OK, I think I should take a look at these new code to find out why the segment fault happens.

**Comment by devbean** [comment]
I think this bug has been fixed in branch release/1.0 . I'm closing this issue. Please reopen it if there is any problem about this.
